apps/ledger-live-desktop/src/newArch/features/ModularDrawer/utils/getAccountTuplesForCurrency.ts

```diff
@@ -0,0 +1,34 @@
+import type { Account, TokenAccount } from "@ledgerhq/types-live";
+import { CryptoCurrency, TokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { makeEmptyTokenAccount } from "@ledgerhq/coin-framework/lib-es/account/helpers";
+
+export type AccountTuple = {
+  account: Account;
+  subAccount: TokenAccount | null;
+};
+
+export function getAccountTuplesForCurrency(
+  currency: CryptoCurrency | TokenCurrency,
+  allAccounts: Account[],
+): AccountTuple[] {
+  const isToken = currency.type === "TokenCurrency";
+
+  return allAccounts
+    .filter(account => {
+      if (isToken) return account.currency.id === currency.parentCurrency.id;
+
+      return account.currency.id === currency.id;
+    })
+    .map(account => {
+      if (isToken) {
+        const subAccount =
+          account.subAccounts?.find(
+            (subAcc: TokenAccount) =>
+              subAcc.type === "TokenAccount" && subAcc.token.id === currency.id,
+          ) || makeEmptyTokenAccount(account, currency);
+
+        return { account, subAccount };
+      }
+      return { account, subAccount: null };
+    });
+}
```

Comment:
```suggestion
export function getAccountTuplesForCurrency(
  currency: CryptoCurrency | TokenCurrency,
  allAccounts: Account[],
): AccountTuple[] {
  const isToken = currency.type === "TokenCurrency";
  const targetCurrencyId = isToken ? currency.parentCurrency.id : currency.id;

  return allAccounts
    .filter(account => account.currency.id === targetCurrencyId)
    .map(account => {
      const subAccount = isToken
        ? account.subAccounts?.find(
            (subAcc): subAcc is TokenAccount =>
              subAcc.type === "TokenAccount" && subAcc.token.id === currency.id,
          ) || makeEmptyTokenAccount(account, currency)
        : null;

      return { account, subAccount };
    });
}
```

This refactor looks solid. A few things that stand out:

Extracting `targetCurrencyId` helps reduce duplication and makes the intent clearer.
Using a ternary for `subAccount` keeps the logic concise and readable.

Nice cleanup overall